DogeNet.WebApi/Controllers/PostController.cs

```diff
@@ -0,0 +1,89 @@
+﻿// <copyright file="PostController.cs" company="Leobit">
+// Copyright (c) Leobit. All rights reserved.
+// </copyright>
+
+namespace DogeNet.WebApi.Controllers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using DogeNet.BLL.Features.Posts;
+    using DogeNet.BLL.Features.Posts.CreatePost;
+    using DogeNet.BLL.Features.Posts.DeletePost;
+    using DogeNet.BLL.Features.Posts.EditPost;
+    using DogeNet.BLL.Features.Posts.GetPost;
+    using DogeNet.DAL;
+    using MediatR;
+    using Microsoft.AspNetCore.Mvc;
+
+    [Route("api/[controller]/[action]")]
+    [ApiController]
+    public class PostController : ControllerBase
+    {
+        private readonly IMediator mediator;
+
+        private readonly DBContext context;
+
+        public PostController(IMediator mediator, DBContext context)
+        {
+            this.mediator = mediator;
+            this.context = context;
+        }
+
+        [HttpGet("{id}")]
+        [ProducesResponseType(200, Type = typeof(PostDetailsModel))]
+        public async Task<IActionResult> GetPostById(int id)
+        {
+            if (PostChecks.PostIdValid(this.context, id))
```

Comment:
This check should be on Validator level

Reply:
but this is get method, there are no model to validate but int id

Reply:
@adrnchk you can validate GetPostQuery and get id from it to validate